src/main/resources/jupyter/safe-mode.js

```diff
@@ -16,87 +16,87 @@ define(() => {
     var googleProject = $(googleProject);
     var clusterName = $(clusterName);
 
-    const welderUrl = `/proxy/${googleProject}/${clusterName}/welder`
-    const checkMetaUrl = welderUrl + '/objects/metadata'
+    const welderUrl = `/proxy/${googleProject}/${clusterName}/welder`;
+    const checkMetaUrl = welderUrl + '/objects/metadata';
 
     const headers = {
         'Content-Type': 'application/json',
-        'Access-Control-Allow-Headers': '*'
-    }
+        'Access-Control-Allow-Headers': '*',
+    };
 
     const basePayload = {
-        mode: "no-cors",
-        headers: headers
-    }
+        mode: 'no-cors',
+        headers: headers,
```

Comment:
Trailing commas are not allowed in the JSON spec http://www.json.org/ (from the discussion here: https://stackoverflow.com/questions/201782/can-you-use-a-trailing-comma-in-a-json-object).

Apparently, some browsers will play nice but others won't, so I think it's best to remove it.

Reply:
I agree with you stylistically - updated.

To clarify though - JSON != Javascript. JSON is stricter than object definition in Javascript, though the naming might lead one to think they are identical. Trailing commas are valid in Javascript, though incidentally they might cause problems in very old browsers like IE8 (already not supported by AoU or Terra). The SO thread you link to is talking about JSON, not Javascript.